### ISSUE TITLE
[Question] What are the .parts files?

### ISSUE THREAD

**Issue by burekas7** [scakemyer/plugin.video.quasar#735]
Sometimes Quasar generates *.parts files in the download directory.
- What are they using for?
- Why are they not deleted automatically? Do they should be deleted?

Thanks.

**Comment by scakemyer** [comment]
It's libtorrent and they do get deleted, if not file an issue over there.

**Comment by burekas7** [comment]
Does it could be because I stop the watching (by press x or stop button) even if the file was 100% downloaded, instead of actually watching it and let it finish naturally?

**Comment by scakemyer** [comment]
No we don't control any of that. You might have some leftover .parts files from previous versions that were setting all file priorities to 0, then setting the one we wanted to normal priority, combined with setting piece priorities, which caused libtorrent to download to .parts files indefinitely. I'd recommend to wait for all your torrents to be at 100%, then ideally close Kodi and remove whatever .parts files are left.

**Comment by burekas7** [comment]
I checked it again, and on each scenario these .parts files don't deleted.
No matter if the download was stopped in the middle, reached to 100%, or if I let it finish naturally after watching.
Currently, I delete them manually on each time.

**Comment by scakemyer** [comment]
Ok I'm seeing this too, so while we got rid of downloading indefinitely to .parts files, my guess is that file priorities don't line up with piece priorities, so even when we reach 100%, then delete the torrents but not the files, libtorrent leaves those .parts file in place. So while I think this should be better handled in libtorrent itself, I'll add something to clean this up when we delete torrents. I did observe a relatively nice side-effect of that with season packs, where starting the next episode ended up being already buffered at around 15% for the initial buffer, which will vary depending on your initial buffer setting obviously, but still. So by deleting the .parts files when we delete torrents only, we'll be cleaning this mess of parts files while getting a little pre-buffering bonus in season packs.

**Comment by burekas7** [comment]
I checked it with v0.9.74, but the .part file still not removed.
I stop the download (It reached to 100%) and chose not keep downloading)

Log: http://pastebin.com/VggbETrG

**Comment by scakemyer** [comment]
Well... `Removing the torrent without deleting files...` so you obviously did choose to keep them, in which case it keeps the .parts files until you do.

**Comment by burekas7** [comment]
I had something weird.
Usually this setting: "Keep Files after playback/download stops" is on.
Previously the .part file was always generated.
Then I turn it off, but the video file was deleted when I stopped the playback, so I turn it on again, and now the .part file isn't generated and the video file isn't deleted.

By the way I had a bug with the setting of "Ask to delete the files after streaming".
When I stopped the playback, it asked me "Delete downloaded file?", I chose "No", but it still removed the video file.

In general, what is the difference between these settings?
"Keep Files after playback/download stops"
"Ask to delete the files after streaming"
